### ISSUE TITLE
Editing existing entry produces Error:API_ERROR: A file with this name already exists

### ISSUE THREAD

**Issue by ValeriyTen** [netlify/netlify-cms#4059]
**Describe the bug**

When I add parameter ```meta: path``` exactly as described in example here - https://www.netlifycms.org/docs/beta-features/#nested-collections, I get an error when editing any existing entry ```Failed to persist entry: API_ERROR: A file with this name already exists```

**To Reproduce**

**Step1:** Add similar collections config in config.yml 
```
collections:
  - name: pages
    label: Pages
    label_singular: 'Page'
    folder: content/pages
    create: true
    # adding a nested object will show the collection folder structure
    nested:
      depth: 100 # max depth to show in the collection tree
      summary: '{{title}}' # optional summary for a tree node, defaults to the inferred title field
    fields:
      - label: Title
        name: title
        widget: string
      - label: Body
        name: body
        widget: markdown
    # adding a meta object with a path property allows editing the path of entries
    # moving an existing entry will move the entire sub tree of the entry to the new location
    meta: { path: { widget: string, label: 'Path', index_file: 'index' } }
```
**Step 2:**  Try to edit ```Body``` of your existing entry in collections then error shows up after pressing ```Publish``` button  

**Expected behavior**

Entry is correctly saved and changes commited to Git repository (or any other in your configuration file)
**Screenshots**

**Applicable Versions:**
- Netlify CMS version: 2.10.52
- Git provider: Gitlab
- OS: MacOS 10.14.6
- Browser version: Google Chrome Version 84.0.4147.89 (official build)

**CMS configuration**

```
backend:
  name: gitlab
  branch: develop
  repo: example/repo
  auth_type: implicit
  app_id: test_app_id
  api_root: https://git.domain.example.com/api/v4
  base_url: https://git.domain.example.com
  auth_endpoint: oauth/authorize
  site-domain: https://docs.domain.example.com/
media_folder: static/images
public_folder: /images
collections:
  - name: pages
    label: Pages
    label_singular: 'Page'
    folder: content
    create: true
    nested:
        depth: 100
        summary: '{{title}}'
    fields:
        - label: Title
          name: title
          widget: string
        - label: Body
          name: body
          widget: markdown
    meta: { path: { widget: string, label: 'Path', index_file: 'index', pattern: ['.*', "Error"] } }
```

**Additional context**

On top of that, parameter 'pattern' seems to be completely ignored, because when ```path``` contains some characters(Uppercase letters, dots etc) it shows error ```USER_PATH IS NOT A VALID PATH```, but pattern allows any character to be included in path

![image](https://user-images.githubusercontent.com/26410032/88180513-504e1a80-cc4f-11ea-8731-c4382ca7c20d.png)

**Comment by ValeriyTen** [comment]
@erezrokah 
Thanks for a fast fix! When can we expect release to test it out on our instance?
I load netlify script from ```unpkg``` CDN? 
Or is there a guide on how to make a package from source?
Thanks!

**Comment by erezrokah** [comment]
This hasn't been released yet (we'll do a release tomorrow probably).
If you'd like to test you can load from `https://deploy-preview-4064--cms-demo.netlify.app/dist/netlify-cms.js`

**Comment by ValeriyTen** [comment]
> If you'd like to test you can load from `https://deploy-preview-4064--cms-demo.netlify.app/dist/netlify-cms.js`

Thanks, @erezrokah! Worked like a charm! I'll wait till release now.

**Comment by erezrokah** [comment]
This is released in `netlify-cms@2.10.55` and `netlify-cms-app@2.12.19`

**Comment by ValeriyTen** [comment]
@erezrokah 

Thanks, it works now!
